### PR TITLE
Upload Controller Refactor

### LINKED ISSUE #22: Refactor upload create action
Refactor bulky upload create action into service, facade and helper methods

### PR DESCRIPTION
**Pull request by nmking22**
* Refactors upload create action into controller/facade/service architecture
* Adds unit tests for new facades
* Adds unit tests for new service
Resolves #22 

## Types of Changes

This PR closes #

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [ ] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [ ] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: app/services/upload_service.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class UploadService
+  def self.upload_audio(audio, key)
+    s3.put_object(bucket: ENV['S3_BUCKET'], key: key, body: audio)
```
Comment (SageOfCode) [inline app/services/upload_service.rb:3]: Great use of a service here, this is awesome.

File: app/controllers/api/v1/upload_controller.rb [hunk 2]
```diff
@@ -97,4 +44,51 @@ def verse_params
   def competition_params
     params.permit(:month, :year, :description, :genre, :rules)
   end
+
+  def has_error?(output)
```
Comment (SageOfCode) [inline app/controllers/api/v1/upload_controller.rb:48]: Great work here on these small methods.

**Review by SageOfCode — APPROVED** [review]
Great work Nick, this is a solid refactor.

File: spec/facades/competition_facade_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    allow_any_instance_of(Aws::S3::Client).to receive(:put_object).and_return(mock_response)
+
+    output = CompetitionFacade.create_competition(competition_params, @audio)
+
+    expect(output).to be_a(Hash)
+    expect(output).to have_key(:error)
+    expect(output[:error]).to eq('Competition could not be successfully uploaded to cloud database.')
+  end
+end
```
Comment (the-color-bliu) [inline spec/facades/competition_facade_spec.rb:64]: Love the TDD, great job

**Review by the-color-bliu — APPROVED** [review]
This refactor looks super clean, merge approved 100%

**Pull request merged**